src/SAWScript/MGU.hs

```diff
@@ -211,32 +210,36 @@ mgus ts1 ts2 =
   failMGU' $ "Wrong number of arguments. Expected " ++ show (length ts1) ++ " but got " ++ show (length ts2)
 
 -- Does not handle the case where t _is_ TyUnifyVar i; the caller handles that
-bindVar :: TypeIndex -> Type -> Either FailMGU Subst
-bindVar i t
-  | i `S.member` unifyVars t = failMGU' "occurs check failMGUs" -- FIXME: error message
-  | otherwise                = return (singletonSubst i t)
+bindVar :: Pos -> TypeIndex -> Type -> Either FailMGU Subst
+bindVar pos i t =
+  case M.lookup i $ unifyVars t of
+     Just otherpos ->
+       -- FIXME/XXX: this error message is better than the one that was here before
+       -- but still lacks
```

Comment:
Is this comment an incomplete sentence?

Reply:
No, but I'll clarify it since it easily appears that way.